File: src/types/Recipe.ts
export interface Recipe {
  id: RecipeId;
  name: string;
  title: string;
  description: string;
  category: string;
  tags: string[];
  image: string;
  prepTime: number;
  cookTime: number;
  servings: number;
  ingredients: Ingredient[];
  steps: Step[];
}

export interface RecipeId {
  timestamp: number;
  creationTime: string;
}

export interface Ingredient {
  quantity: number;
  unit: string;
  name: string;
}

export interface Step {
  step: number;
  instruction: string;
}

File: backend/Models/Recipe.cs
using MongoDB.Bson;

namespace BackendAPI.Models
{
    public class Recipe
    {
        public ObjectId Id { get; set; } // this maps to _id in mongo db atlas by default
        public string Name { get; set; } = String.Empty;
        public string Title { get; set; } = String.Empty;
        public string Description { get; set; } = String.Empty;
        public string Category { get; set; } = String.Empty;

        public List<string>? Tags { get; set; }
        public string Image { get; set; } = String.Empty;
        public int PrepTime { get; set; }
        public int CookTime { get; set; }
        public int Servings { get; set; }

        public List<Ingredient>? Ingredients { get; set; }
        public List<Step>? Steps { get; set; }



    }
}